### DISCUSSION TITLE
[Bug] Dynamic type for methods doesnt support auto conversion.

### DISCUSSION THREAD

**Discussion by faruknane** [dotnet/csharplang#3800] [Q&A]
I think there is a bug (or this is a feature request) that should be fixed to enable us to use the dynamic type more python-like.

![image](https://user-images.githubusercontent.com/37745467/90345292-07a83800-e028-11ea-8b98-8491e58cf5d5.png)

Normally, Func<string> typed object can be converted to dynamic type. However, when I try to convert a function to dynamic type, it throws a compile-time error.  

But why is that?

Is it because it can't directly convert it to dynamic type, it needs an intermediate step first function to Func<string> then to dynamic type?

I think this is a small issue, but should be fixed.

**Comment by HaloFour** [comment]
Method groups aren't a type in C# and you can't capture them in a variable without an explicit delegate type.  Even ignoring issues with overloading there would still be any number of potential delegate types that could match the signature of that method.  To make this seem more like Python you'd probably have to have the compiler emit some synthetic type that wraps all of the possible methods and binds to them at runtime.   Probably not worth the effort.

**Comment by faruknane** [comment]
Thank you for your answer. @HaloFour

**Comment by HaloFour** [comment]
You can leave this issue open.  I'm just giving some background as to why you probably can't use a method group directly as a `dynamic` variable (which is similar to why you can't use it with `var`) and my opinion what could be done about it.  But I'm not a member of the language team.
